shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPlugin.mm

```diff
@@ -0,0 +1,104 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPlugin.h"
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIApplication.h>
+#import <UIKit/UIKit.h>
+
+#import "flutter/fml/logging.h"
+#import "flutter/shell/platform/darwin/ios/framework/Source/FlutterViewController_Internal.h"
+
+// Method Channel name to start spell check.
+static NSString* const kInitiateSpellCheck = @"SpellCheck.initiateSpellCheck";
+
+@interface FlutterSpellCheckPlugin ()
+
+@property(nonatomic, assign) FlutterMethodChannel* methodChannel;
+@property(nonatomic, retain) UITextChecker* textChecker;
+
+@end
+
+@implementation FlutterSpellCheckPlugin
+
+- (instancetype)initWithMethodChannel:(FlutterMethodChannel*)methodChannel {
+  self = [super init];
+  if (self) {
+    [_methodChannel setMethodCallHandler:^(FlutterMethodCall* call, FlutterResult result) {
+      [self handleMethodCall:call result:result];
+    }];
+    _textChecker = [[[UITextChecker alloc] init] retain];
+  }
+  return self;
+}
+
+- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
+  NSString* method = call.method;
+  id args = call.arguments;
+  if ([method isEqualToString:kInitiateSpellCheck]) {
+    FlutterError* error;
+    NSDictionary* spellCheckResult = [self initiateSpellCheck:args withError:&error];
+    if (error) {
+      result(error);
+    } else {
+      result(spellCheckResult);
+    }
+  }
+}
+
+// Get the misspelled word and suggestions.
+//
+// The result will be formatted as a NSDictionary that looks like this:
+// {
+//   @"startIndex": 0,
+//   @"endIndex" : 5,
+//   @"suggestions": [@"suggestion1", @"suggestion2"..]
+// }
+- (NSDictionary*)initiateSpellCheck:(id)args withError:(FlutterError**)error {
```

Comment:
```suggestion
- (NSDictionary<NSString *, id>*)initiateSpellCheck:(id)args withError:(FlutterError**)error {
```

Reply:
Actually this name is kind of confusing, it's not like it's kicking off a spell check to be processed later.

How about something like `spellCheckSuggestionsForText:inLanguage:withError:` where the arg indexing happens one layer up?